src/validation.h

```diff
@@ -587,8 +587,9 @@ class CChainState
      */
     mutable std::atomic<bool> m_cached_finished_ibd{false};
 
-    //! mempool that is kept in sync with the chain
-    CTxMemPool& m_mempool;
+    //! Optional mempool that is kept in sync with the chain.
+    //! Only the active chainstate has a mempool.
+    CTxMemPool* m_mempool;
```

Comment:
This could be a const pointer:

```suggestion
    CTxMemPool* const m_mempool;
```

since it's set in the initializer list and never modified.

Reply:
The pointer may actually be nulled out e.g. when we transition the initial IBD chainstate to the background (and give its mempool to the snapshot chainstate, which becomes our active).

Reply:
ah, ok! I'm not totally up to date on how the different chainstate objects work in assumeutxo.